### PR TITLE
feat: Support `'latest'` as `ecmaVersion`

### PR DESCRIPTION
**Pull request by wafuwafu13**
Done: https://github.com/typescript-eslint/typescript-eslint/issues/3857

### CODE REVIEW

File: packages/types/src/parser-options.ts [hunk 1]
```diff
@@ -26,7 +30,7 @@ interface ParserOptions {
     globalReturn?: boolean;
     jsx?: boolean;
   };
-  ecmaVersion?: EcmaVersion;
+  ecmaVersion?: EcmaVersion | 'latest';
```
Comment (wafuwafu13) [inline packages/types/src/parser-options.ts:33]: https://github.com/typescript-eslint/typescript-eslint/pull/3710#discussion_r698641593
>let's remove latest from here and add it to the parser types as we're now handling it specifically within the parser

File: packages/types/src/parser-options.ts [hunk 2]
```diff
@@ -12,12 +12,16 @@ type EcmaVersion =
...
+  | 13
   | 2015
   | 2016
   | 2017
   | 2018
   | 2019
-  | 2020;
+  | 2020
+  | 2021
+  | 2022;
```
Comment (wafuwafu13) [inline packages/types/src/parser-options.ts:24]: https://github.com/typescript-eslint/typescript-eslint/pull/3710#discussion_r698455250
>Added 2021 and 2022

File: packages/scope-manager/src/analyze.ts [hunk 3]
```diff
@@ -81,7 +82,11 @@ const DEFAULT_OPTIONS: Required<AnalyzeOptions> = {
   emitDecoratorMetadata: false,
 };
 
-function mapEcmaVersion(version: EcmaVersion | undefined): Lib {
+/**
+ * Convert ecmaVersion to lib.
+ * `'latest'` is converted to 1e8 at parser.
+ */
+function mapEcmaVersion(version: EcmaVersion | 1e8 | undefined): Lib {
```
Comment (wafuwafu13) [inline packages/scope-manager/src/analyze.ts:89]: Ref: https://github.com/typescript-eslint/typescript-eslint/pull/3710#discussion_r698642357
>you don't need/want this check any more

**Review by bradzacher — APPROVED** [review]
thanks!

**Pull request merged**